libraries/AP_HAL_PX4/AnalogIn.cpp

```diff
@@ -380,14 +380,16 @@ void PX4AnalogIn::_timer_tick(void)
 
 AP_HAL::AnalogSource* PX4AnalogIn::channel(int16_t pin) 
 {
-    for (uint8_t j=0; j<PX4_ANALOG_MAX_CHANNELS; j++) {
+    uint8_t j;
+    for (j=0; j<PX4_ANALOG_MAX_CHANNELS; j++) {
         if (_channels[j] == nullptr) {
```

Comment:
See here.

Reply:
that is not indexed by pin. I don't see what stops it allocating every time

Reply:
It only allocates if the item at the current index is null. There are only PX4_ANALOG_MAX_CHANNELS elements, this is the maximum AnalogSources that will be allocated.

Reply:
ok, and when that fails then the adc send function in sub will crash with a null ptr dereference

Reply:
It never returns null.